adoc/headers/queue.h

```diff
@@ -177,16 +147,48 @@ class queue {
             access::placeholder IsSrcPlaceholder, typename DestT, int DestDims,
             access_mode DestMode, target DestTgt,
             access::placeholder IsDestPlaceholder>
-  event
-  copy(accessor<SrcT, SrcDims, SrcMode, SrcTgt, IsSrcPlaceholder> src,
-       accessor<DestT, DestDims, DestMode, DestTgt, IsDestPlaceholder> dest);
+  event copy(accessor<SrcT, SrcDims, SrcMode, SrcTgt, IsSrcPlaceholder> src,
+             accessor<DestT, DestDims, DestMode, DestTgt, IsDestPlaceholder> dest);
+
+  /* -- Shortcut functions: memset -- */
+
+  event memset(void* ptr, int value, size_t numBytes);
+  event memset(void* ptr, int value, size_t numBytes, event depEvent);
+  event memset(void* ptr, int value, size_t numBytes,
+               const std::vector<event>& depEvents);
+
+  /* -- Shortcut functions: fill -- */
+
+  template <typename T>
+  event fill(void* ptr, const T& pattern, size_t count);
+  template <typename T>
+  event fill(void* ptr, const T& pattern, size_t count, event depEvent);
+  template <typename T>
+  event fill(void* ptr, const T& pattern, size_t count,
+             const std::vector<event>& depEvents);
 
   template <typename T, int Dims, access_mode Mode, target Tgt,
             access::placeholder IsPlaceholder>
-  event update_host(accessor<T, Dim, Mode, Tgt, IsPlaceholder> acc);
+  event fill(accessor<T, Dims, Mode, Tgt, IsPlaceholder> dest, const T& src);
+
+  /* -- Shortcut functions: prefetch -- */
+
+  event prefetch(void* ptr, size_t numBytes);
+  event prefetch(void* ptr, size_t numBytes, event depEvent);
+  event prefetch(void* ptr, size_t numBytes,
+                 const std::vector<event>& depEvents);
+
+  /* -- Shortcut functions: mem_advise -- */
+
+  event mem_advise(void* ptr, size_t numBytes, int advice);
+  event mem_advise(void* ptr, size_t numBytes, int advice, event depEvent);
+  event mem_advise(void* ptr, size_t numBytes, int advice,
+                   const std::vector<event>& depEvents);
+
+  /* -- Shortcut functions: update_host -- */
 
   template <typename T, int Dims, access_mode Mode, target Tgt,
             access::placeholder IsPlaceholder>
-  event fill(accessor<T, Dims, Mode, Tgt, IsPlaceholder> dest, const T& src);
+  event update_host(accessor<T, Dim, Mode, Tgt, IsPlaceholder> acc);
```

Comment:
Just realized it was a mistake. Should be `Dims`. `Dim` not defined in the template.

Reply:
Good catch.  Fixed in 46907d4df3c98edad84fa1ab3e68b6f6b419cc65